src/gurt/hash.c

```diff
@@ -1050,7 +1053,7 @@ rl_op_empty(struct d_rlink *rlink)
 		return true;
 
 	is_unlinked = d_hash_rec_unlinked(&rlink->rl_link);
-	D_ASSERT(rlink->rl_ref != 0 || is_unlinked);
+	D_ASSERT(atomic_load_relaxed(&rlink->rl_ref) != 0 || is_unlinked);
```

Comment:
I'm wondering is this is entirely right, the atomics in this file are correctly using memory order relaxed so the only synchronisation will be against other accesses to the same atomic or through general memory barrier which this code itself probably does not provide (but is likely to happen by any code that calls this), would it be safer to write this as:
```
bool is_unlinked = (atomic_load_relaxed(&rlink->rl_ref == 0);
if (is_unlinked)
  D_ASSERT(d_hash_rec_unlinked(&rlink->rl_link);
return is_unlinked;
```
They're probably exactly the same in practice although there would be fewer memory accesses in the second one for the common case.